script.js

```diff
@@ -1,4 +1,159 @@
+    allDiceOrder.push(diceOrder);
+    console.log(allDiceOrder);
+  }
+
+  if (accumulativeRound >= 1) {
+    if (currentPlayer == 1) {
+      allDiceOrder[0] = allDiceOrder[0] + diceOrder;
+      console.log(`new array value = ${allDiceOrder[0]}`);
+      diceOrder = [];
+    }
+    if (currentPlayer == 2) {
+      allDiceOrder[1] = allDiceOrder[1] + diceOrder;
+      console.log(`new array value = ${allDiceOrder[0]}`);
+      diceOrder = [];
+    }
+    playerRoll1 = [];
+    playerRoll2 = [];
+  }
+  return myOutputValue;
+};
+
+//functiont to determine the winner by comaring the values in the array
+var determineWinner = function () {
+  var myOutputValue = `Player 1 score is ${allDiceOrder[0]}<br>
+    Player 2 score is ${allDiceOrder[1]}<br>`;
+  var endOfSentence = `This is not the end! Click submit again to continue the game and try to beat your friend!  `;
+  if (allDiceOrder[0] >= allDiceOrder[1]) {
+    myOutputValue =
+      myOutputValue +
+      `<br><br><b>The winner is Player 1.</b><br><br>` +
+      endOfSentence;
+  }
+  if (allDiceOrder[1] >= allDiceOrder[0]) {
+    myOutputValue =
+      myOutputValue +
+      `<br><br> <b>The winner is Player 2!</b><br><br>` +
+      endOfSentence;
+  }
+  if (allDiceOrder[1] == allDiceOrder[0]) {
+    myOutputValue =
+      myOutputValue + `<br><br> <b>It is a draw!</b><br><br>` + endOfSentence;
+  }
+
   return myOutputValue;
 };
+
+//function for the game
+var beatThatGame = function (input) {
+  //first mode is when the player rolls 2 dice
+  if (gameMode == "start game by rolling 2 dice") {
+    console.log(gameMode);
+    var diceRoll = randomDiceGenerator();
+    //switch mode to choose dice order
+    gameMode = "player choose dice order";
+    console.log(gameMode);
+    return diceRoll;
+  }
+  if (gameMode == "player choose dice order") {
+    var diceScore = chooseDiceOrder(input);
+    return diceScore;
+  }
+};
+
+var main = function (input) {
+  if (gameMode == "next player roll 2 dice") {
+    console.log(gameMode);
+    //loop to have more than 2 or as many players in the game.
+    while (currentPlayer < 2) {
+      currentPlayer += 1;
+      gameMode = "start game by rolling 2 dice";
+      console.log(gameMode);
+      return `It is now Player ${currentPlayer}'s turn! <br><br> Please click submit to roll the dice.`;
+    }
+    //after 3 players have played, change game mode to compare scores
+    if (currentPlayer == 2) {
+      console.log(`end of last player's turn`);
+      gameMode = "determine winner among 2 players";
+      return `This is the end of the game.<br><br> Please click submit to see who is the winner. `;
+    }
+  }
+  if (gameMode == "determine winner among 2 players") {
+    console.log(gameMode);
+    var winner = determineWinner();
+    //game continues indefinitely
+    gameMode = "start game by rolling 2 dice";
+    currentPlayer = 1;
+    accumulativeRound = 1;
+    return winner;
+  }
+  return beatThatGame(input);
+};
```

Comment:
i think your `main` and `beatThatGame` functions are sufficiently succinct that they can be merged together, this way the code becomes more readable too, as there is less jumping around between different functions and the single function will outline the game flow by nature of the order of the code